sw/device/lib/crypto/drivers/kmac.c

```diff
@@ -656,3 +665,107 @@ kmac_error_t kmac_process_msg_blocks(kmac_operation_t operation,
 
   return kKmacOk;
 }
+
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_sha3_224(const uint8_t *data, size_t data_len,
+                           uint8_t *digest) {
+  if (data == NULL && data_len > 0) {
+    return kKmacArgsError;
+  } else if (digest == NULL) {
+    return kKmacArgsError;
+  }
+
+  kmac_error_t err;
+  err = kmac_init(kKmacOperationSHA3, kKmacSecurityStrength224,
+                  /*func_name=*/NULL, /*func_name_len=*/0,
+                  /*cust_str=*/NULL, /*cust_str_len=*/0);
+  if (err != kKmacOk) {
+    return err;
+  }
+
+  err = kmac_process_msg_blocks(kKmacOperationSHA3, data, data_len, digest,
+                                /*digest_len=*/224 / 8);
```

Comment:
`data` & `data_len`, `digest` & `digest_len` pair suggests that we should consider defining a generic buffer type like:
```
typedef struct crypto_buf_u8 {
  uint8_t * data,
  size_t len,
} crypto_buf_u8_t;
```